internal/util/testutil/equal.go

```diff
@@ -174,6 +175,9 @@ func equalScalars(tb testing.TB, v1, v2 any) bool {
 		if math.IsNaN(s1) {
 			return math.IsNaN(s2)
 		}
+		if s1 == 0 && s2 == 0 {
+			return math.Signbit(s1) == math.Signbit(s2)
+		}
```

Comment:
why not: 
math.Signbit(s1) == math.Signbit(s2) && s1 == s2?

Reply:
it will be returning `equality` even if signs are not equal, according to the code below. So maybe it doesn't make sense. WDYT?

Reply:
Assuming you mean

```go
if math.Signbit(s1) == math.Signbit(s2) && s1 == s2 {
  return true
}

return s1 == s2
```

that will not pass `AssertNotEqual(t, math.Copysign(0, +1), math.Copysign(0, -1))` test: condition under `if` will return `false`, but `==` is defined to return `true` for `math.Copysign(0, +1) == math.Copysign(0, -1)`. That's exactly why we need that PR at all

Reply:
ok, that unexpected behavior of `AssertNotEqual`

Reply:
What's unexpected about it?

Reply:
`AssertNotEqual(t, math.Copysign(0, +1), math.Copysign(0, -1))` passes with merged changes (values are not equal). Suggest change (`math.Signbit(s1) == math.Signbit(s2) && s1 == s2`) would fail that test